python/lsst/ts/observatory/control/utils/extras/guider_roi.py

```diff
@@ -0,0 +1,790 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Guider ROI selection for LSSTCam.
+
+This module implements guider ROI selection functionality based on the
+original algorithm developed by Aaron Roodman. The implementation has been
+adapted for integration with ts_observatory_control.
+
+Original algorithm credit: Aaron Roodman
+"""
+
+__all__ = [
+    "BASIC_DEPS_AVAILABLE",
+    "DM_STACK_AVAILABLE",
+    "BEST_EFFORT_ISR_AVAILABLE",
+    "GuiderROIs",
+    "get_vignetting_data_from_butler",
+]
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from lsst.summit.utils import BestEffortIsr as BestEffortIsrType
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
```

Comment:
Please, remove this logger here, use the `warn` package to warn instead of `log`. See [here](https://github.com/lsst-ts/ts_observatory_control/blob/eff49dd43f4a23477a5703d5aa55ac9220969f41/python/lsst/ts/observatory/control/utils/extras/dm_target_catalog.py#L41-L44).